Firestore/core/src/firebase/firestore/local/leveldb_persistence.cc

```diff
@@ -123,7 +123,8 @@ LevelDbPersistence::LevelDbPersistence(std::unique_ptr<leveldb::DB> db,
 // MARK: - Storage location
 
 util::Path LevelDbPersistence::StorageDirectory(
-    const core::DatabaseInfo& database_info, const util::Path& documents_dir) {
+    const core::DatabaseInfo& database_info,
+    const util::StatusOr<util::Path>& documents_dir) {
```

Comment:
The caller should handle the status not being OK and only call StorageDirectory with a valid Path. That way we can centralize how we fail in response to errors of any kind preventing the creation of the persistence instance.

Reply:
Reverted.